### ISSUE TITLE
C++20 concepts supported by doxygen v1.9.2+

### ISSUE THREAD

**Issue by 2bndy5** [michaeljones/breathe#811]
I've been hunting errors in the docs build logs, and I found that the cpp-concept example (located in examples/specific) was not being properly parsed by doxygen in the CI. It is parsed as expected in doxygen v1.9.2 (&1.9.3) per local testing.

My error hunting goal is to make sphinx treat a warning as an error, thus making the doc building CI runs fail when new warnings have not been addressed. Due to this issue, we'd have to pin CI runs to doxygen v1.9.2 or newer (which can be done per #782 ). There is also a way to use conda forge to make RTD builds use a specific version of doxygen (despite the runner's frozen PPAs) which is a simple change in readthedocs.yaml.

**Comment by vermeeren** [comment]
@2bndy5 I think in such cases it's fine to skip specific tests if they are known to be broken with specific Doxygen versions.

**Comment by 2bndy5** [comment]
Unfortunately, This test is baked into the docs. I suppose we could do an RST `.. ifconfig::` like we do with the version warning in index,rst. That would allow using any version of Doxygen to build the docs.

You see how feedback helps me develop a better approach?

**Comment by vermeeren** [comment]
> Unfortunately, This test is baked into the docs. I suppose we could do an RST `.. ifconfig::` like we do with the version warning in index,rst. That would allow using any version of Doxygen to build the docs.

@2bndy5 Good idea, that should definitely make it do-able. According to doc https://www.sphinx-doc.org/en/master/usage/extensions/ifconfig.html it uses a generic "python expression", so it should be possible to have actual version comparisons.

For example, by setting the Doxygen version used as a variable and depending on the version `>=` etc dynamically enable/disable the parts as needed.

> You see how feedback helps me develop a better approach?

Yes? (Maybe it's just my fried brain today but not sure if I understand what you mean here.)

**Comment by vermeeren** [comment]
@2bndy5 Ah yeah, that's definitely true! Thanks again for all your help recently!

**Comment by 2bndy5** [comment]
Ok, I've implemented the `ifconfig` idea, but I still think the breathe docs should be built with a version of doxygen that can handle C++20 concepts (1 of the changes in #816)
